tests/testthat/helper-mock-cli.R

```diff
@@ -0,0 +1,22 @@
+real_wcr <- wsl_compatible_run
+
+with_mocked_cli <- function(code, compile_ret, info_ret){
+  with_mocked_bindings(
+    code,
+    wsl_compatible_run = function(command, args, ...) {
+      if (
+        !is.null(command)
+        && command == 'make'
+        && !is.null(args)
+        && startsWith(basename(args[1]), 'model-')
+      ) {
+        message("mock-compile-was-called")
+        compile_ret
+      } else if (!is.null(args) && args[1] == "info") info_ret
+      else real_wcr(command = command, args = args, ...)
+    }
+  )
+}
+
+expect_mock_compile <- function(object, ...) expect_message(object, regexp = 'mock-compile-was-called', ...)
+expect_no_mock_compile <- function(object, ...) expect_no_message(object, message = 'mock-compile-was-called' , ...)
```

Comment:
Overall what I'm trying accomplish here is basically mimic python's `assert_called` and `assert_not_called`. (https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called).

**Logic is**: `expect_mock_compile` will pass if mock_compile is called (at all, doesn't matter how many times) and fail if mock_compile is never called. `expect_no_mock_compile` is the inverse. It passes if mock_compile is *not* called at all and fails if mock_compile is called (even once).

**Implementation is**: `mock_compile` emits a message with the contents `mock-compile-was-called`. `expect_mock_compile` checks for this message: passes if it detects such a message, fails if it does not. `expect_no_mock_compile` fails if a message with exactly this text is detected and passes otherwise. A message with any other text does not impact `expect_no_mock_compile`.

There is probably a more elegant way to achieve this. Open to suggestions.

Reply:
> There is probably a more elegant way to achieve this. Open to suggestions.

There may be, but I can't think of one at the moment. I would go ahead with this implementation, unless @SteveBronder has a better idea.

Reply:
That's fine thanks for the clarification. Can you add a little comment (the above) just so people in the future can know what to expect?

Reply:
✅